animation_nodes/nodes/object/lamp_output.py

```diff
@@ -25,5 +25,16 @@ def create(self):
             ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
             ("Objects", "objects")))
 
+        for socket in self.inputs[1:]:
+            socket.useIsUsedProperty = True
+            socket.isUsed = False
+
     def getExecutionCode(self, required):
-        return "if object is not None and object.type == 'LIGHT': object.data.color = color[:3]; object.data.energy = energy"
+        s = self.inputs
+        isColor = s[1].isUsed
+        isEnergy = s[2].isUsed
+
+        if any([isColor, isEnergy]):
+            yield "if object is not None and object.type == 'LIGHT':"
+            if isColor:    yield "        object.data.color = color[:3]"
+            if isEnergy:   yield "        object.data.energy = energy"
```

Comment:
Those looks like 8 spaces of indentation instead of 4? Does it work like this?

Reply:
Yes. Because of the previous if statement.